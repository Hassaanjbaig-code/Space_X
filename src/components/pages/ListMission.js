import React from 'react'

const ListMission = () => {
  return (
    <div>
      <h1>ListMission</h1>
    </div>
  )
}

export default ListMission
