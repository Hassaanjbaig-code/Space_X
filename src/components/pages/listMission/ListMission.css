.Mission-List {
  list-style: none;
  display: grid;
  grid-template-columns: 11rem 63% 12% 12%;
}

.Mission-List p {
  text-align: justify;
  padding: 10px;
}

.Header {
  font-weight: 900;
}

.border {
  border: 1px solid #e5e6eb;
}

.Mission-row h3 {
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Not-Approved {
  background: #6d757d;
  border-radius: 10px;
  color: #fff;
}

.Not-Approved-button {
  border: 1px solid #121212;
  padding: 11px;
  font-size: 1rem;
  background: #fff;
  border-radius: 9px;
  cursor: pointer;
}

.Not-Approved-button:active {
  background-color: rgba(255, 0, 0, 0.403);
}
