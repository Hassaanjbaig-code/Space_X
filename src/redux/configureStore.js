import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketReducer';
import missionReducer from './missions/missionReducer';

export default configureStore({
  reducer: {
    rocketReducer,
    mission: missionReducer.reducer,
  },
});
