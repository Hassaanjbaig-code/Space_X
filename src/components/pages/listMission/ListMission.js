import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { missionaction } from '../../../redux/missions/missionReducer';
import './ListMission.css';

const ListMission = () => {
  const dispatch = useDispatch();
  const show = useSelector((data) => data.mission);
  useEffect(() => {
    dispatch(missionaction());
  }, [dispatch]);
  return (
    <div>
      <ul className="List-mission-main">
        <li className="Mission-List">
          <h3 className="border">Mission</h3>
          <p className="Header border">Description</p>
          <h4 className="border">Status</h4>
        </li>
        {show.data.map((data) => (
          <li className="Mission-List Mission-row" key={data.id} id={data.id}>
            <h3 className="border">{data.name}</h3>
            <p className="border">{data.description}</p>
            {data.reserved ? (
              <div className="border flex-center">
                <p>Active Menmber</p>
              </div>
            ) : (
              <div className="border flex-center">
                <p className="Not-Approved">NOT A MEMBER</p>
              </div>
            )}
            {data.reserved ? (
              <div className="border flex-center">
                <button type="button" className="Approved-button">Join Mission</button>
              </div>
            ) : (
              <div className="border flex-center">
                <button type="button" className="Not-Approved-button">Leave Mission</button>
              </div>
            )}
          </li>
        ))}
      </ul>
    </div>
  );
};

export default ListMission;
