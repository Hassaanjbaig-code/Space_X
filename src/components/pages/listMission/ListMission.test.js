import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import React from 'react';
import ListMission from './ListMission';
import configureStore from '../../../redux/configureStore';

const store = configureStore;
describe('teting', () => {
  test('should ', () => {
    jest.mock('../../../redux/missions/missionReducer.js');
    const tree = (
      <React.StrictMode>
        <Provider store={store}>
          <ListMission />
        </Provider>
      </React.StrictMode>
    );
    expect(tree).toMatchSnapshot();
  });
});
