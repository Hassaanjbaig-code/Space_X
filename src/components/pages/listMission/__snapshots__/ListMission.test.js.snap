// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teting should  1`] = `
<UNDEFINED>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <ListMission />
  </Provider>
</UNDEFINED>
`;
